src/vos/vos_pool.c

```diff
@@ -734,6 +711,92 @@ vos_pool_open(const char *path, uuid_t uuid, bool small, daos_handle_t *poh)
 	return rc;
 }
 
+int
+vos_pool_open(const char *path, uuid_t uuid, unsigned int flags,
+	      daos_handle_t *poh)
+{
+	struct vos_pool_df	*pool_df;
+	struct vos_pool		*pool = NULL;
+	struct d_uuid		 ukey;
+	PMEMobjpool		*ph;
+	int			 rc, enabled = 1;
+
+	if (path == NULL || poh == NULL) {
+		D_ERROR("Invalid parameters.\n");
+		return -DER_INVAL;
+	}
+
+	uuid_copy(ukey.uuid, uuid);
+	D_DEBUG(DB_MGMT, "Pool Path: %s, UUID: "DF_UUID"\n", path,
+		DP_UUID(uuid));
+
+	if (flags & VOS_POF_SMALL)
+		flags |= VOS_POF_EXCL;
```

Comment:
Ken, does this implication make sense to you? I'm not so sure I understand the implementation of "small" correctly.

Reply:
Not necessarily. Is the intention here to add a convenience for callers such as rdb so that they won't have to specify both VOS_POF_SMALL and VOS_POF_EXCL? But I think I saw elsewhere in this patch that the rdb callers are explicitly providing both flags.

Small was added after a vos change caused potential problems for the rdb use case. The vos space reservation formula had changed so that its calculated SCM reserved size within the vos pool be no smaller than 2GB (and roughly less than 50% of the total vos pool size). But the rdb allocations are typically only 128MB or perhaps 1GB depending on setting of DAOS_MD_CAP. And the impact of turning off space reservation is that background operations like aggregation, GC can be negatively affected (could hurt rdb log compaction). So the small flag is asking vos to use its original space reservation formula for such small pools.

Reply:
> Is the intention here to add a convenience for callers such as rdb so that they won't have to specify both VOS_POF_SMALL and VOS_POF_EXCL?

The intention of the implication is to prevent

    vos_pool_open(pool X, SMALL but no EXCL);
    vos_pool_open(pool X, no SMALL and no EXCL); /* will succeed but in SMALL mode? */

and

    vos_pool_open(pool X, no SMALL and no EXCL);
    vos_pool_open(pool X, SMALL and no EXCL); /* will succeed but not in SMALL mode? */

Do I understand it correctly?